sdk/messaging/azservicebus/internal/namespace.go

```diff
@@ -287,19 +317,23 @@ func (ns *Namespace) Recover(ctx context.Context, theirConnID uint64) (bool, err
 
 // negotiateClaim performs initial authentication and starts periodic refresh of credentials.
 // the returned func is to cancel() the refresh goroutine.
-func (ns *Namespace) NegotiateClaim(ctx context.Context, entityPath string) (func() <-chan struct{}, error) {
+func (ns *Namespace) NegotiateClaim(ctx context.Context, entityPath string) (context.CancelFunc, <-chan struct{}, error) {
```

Comment:
I don't think this needs to be split into three return args.  Looks like the bug was introduced in the track 2 migration.  See the [track 1](https://github.com/Azure/azure-service-bus-go/blob/master/namespace.go#L293-L296) impl where we closed the channel when the refresh goroutine exits.

Reply:
This change was less about a bug and more about just wanting to make it simpler to do the common operations. Basically, when I cancel I never had a case where I intentionally was trying to do it and walk way - I always wanted to block and wait until the refresh loop exits.

I also had a use case, in tests, where I wanted to make sure that the refresh loop exits without me explicitly cancelling it (ie, fatal recovery error).

So there's some internal expedience, and also just making it simpler to use it in the way that I actually used it.

Reply:
With the old impl, you would write `<-cancelFun()` to wait until the exit channel was closed (so it wasn't fire-and-forget).  But you have a good point about wanting to detect the refresh goroutine exiting due to a terminal failure, so I can see the sense in breaking this apart.